### PR TITLE
Bug 1946232: VM List can consume very high CPU

### PR DESCRIPTION
**Pull request by glekner**
This PR adds a lazy status that loads the vm status bundle after the table renders.
This change improves the performance of queries when having 500+ VMs

While the status loads almost instantly, this is how the loading component looks for the status column

https://user-images.githubusercontent.com/24938324/127858961-fc55bc4f-cb14-43b1-94b5-360e8cc20c75.mov

-----------------------------

**Here are performance recordings on x4 CPU slowdown, from the moment a query was typed until the new results were rendered:**

Before 9033ms
Screen Shot 2021-08-02 at 14 54 44

After 4601ms
Screen Shot 2021-08-02 at 14 48 58

### CODE REVIEW

**Comment by glekner** [comment]
@rawagner @vojtechszocs @yaacov PTAL

File: frontend/packages/kubevirt-plugin/src/components/vm-status/lazy-vm-status.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        pods,
+        migrations,
+        pvcs,
+        dataVolumes: dvs,
+        vmImports: [],
+      });
+      setVmStatus(vmStatusBundle);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loaded]);
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/src/components/vm-status/lazy-vm-status.tsx:52]: so the vm status is never updated beside the first one when you load resources ? That does not seem correct

File: frontend/packages/kubevirt-plugin/src/components/vm-status/use-vm-status-resources.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+export const useVmStatusResources = (namespace: string): VmStatusResourcesValue => {
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [loadError, setLoadError] = React.useState<string>('');
+  const [pods, setPods] = React.useState<PodKind[]>([]);
+  const [migrations, setMigrations] = React.useState<K8sKind[]>([]);
+  const [dvs, setDvs] = React.useState<V1alpha1DataVolume[]>([]);
+  const [pvcs, setPvcs] = React.useState<PersistentVolumeClaimKind[]>([]);
+
+  const watchedResources = React.useMemo(
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/src/components/vm-status/use-vm-status-resources.tsx:33]: you do not have to use useMemo anymore, watchK8s hook is comparing the input args using deep compare

File: frontend/packages/kubevirt-plugin/src/components/vm-status/lazy-vm-status.tsx (hunk 1 above)
Comment (glekner) [reply frontend/packages/kubevirt-plugin/src/components/vm-status/lazy-vm-status.tsx:52]: you are right, I guess there is no going around listening to all resources? @rawagner

File: frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx [hunk 3]
```diff
@@ -125,27 +116,25 @@ const VMHeader = (t: TFunction) => () =>
...
-  if (vmImport) {
-    model = VirtualMachineImportModel;
-    options = vmImportMenuActions.map((action) => action(model, vmImport));
-  } else if (vm) {
+  if (vm) {
     model = VirtualMachineModel;
     options = vmMenuActions.map((action) =>
       action(model, vm, {
-        vmStatusBundle,
+        vmStatusBundle: {},
```
Comment (glekner) [inline frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx:137]: @rawagner do you think creating a lazy kebab menu is a viable solution here?

File: frontend/packages/kubevirt-plugin/src/components/vm-status/lazy-vm-status.tsx (hunk 1 above)
Comment (rawagner) [reply frontend/packages/kubevirt-plugin/src/components/vm-status/lazy-vm-status.tsx]: there isnt. You can optimize it by using `useMemo` and not the state/effect combo.
You can also try comparing previous state result with current one with deep compare

File: frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx (hunk 3 above)
Comment (rawagner) [reply frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx:137]: hm...maybe let the user know you're waiting for status ? Something like disabled item `loading status` etc ?

**Comment by glekner** [comment]
- Added `LazyVmRowKebab` @rawagner

File: frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx [hunk 4]
```diff
@@ -279,157 +260,69 @@ const VirtualMachinesPage: React.FC<VirtualMachinesPageProps> = (props) => {
...
+        metadata: {
+          name: getName(obj),
+          namespace: getNamespace(obj),
+          creationTimestamp: getCreationTimestamp(obj),
+          node: getVMINodeName(objectBundle.vmi),
+          uid: getUID(obj),
+          lookupID,
+          labels: getLabels(obj),
+          // printableStatus is only available for latest kubevirt version, so if its not available we sort by ready
+          status: obj?.status?.printableStatus || obj?.status?.ready,
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx:321]: where is the fallback to vm-status method when  obj?.status?.printableStatus is not available

Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx:321]: /hold
`obj?.status?.ready` is not the currect fallbak is should be` vmStatusBundle.status.toSimpleSortString()`

**Pull request merged**